src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ADRController.kt

```diff
@@ -205,16 +207,42 @@ class ADRController(private val encryption: Encryption,
                 listOf("name" to resourceFileName, "description" to artefact.second, "hash" to file.md5sum(),
                         "resource_type" to resourceType)
         val adr = adrClientBuilder.build()
+        val newDatasetHash = commonParameters[2].second
         return try
         {
             when (resourceId)
             {
                 null -> adr.postFile("resource_create", commonParameters + listOf("package_id" to id), filePart)
-                else -> adr.postFile("resource_patch", commonParameters + listOf("id" to resourceId), filePart)
+                else ->
+                {
+                    val fileHasNoChanges = uploadFileHasNoChanges(resourceId, newDatasetHash)
+                    if (fileHasNoChanges)
+                    {
+                        SuccessResponse(null).asResponseEntity()
+                    }
+                    else
+                    {
+                        adr.postFile("resource_patch", commonParameters + listOf("id" to resourceId), filePart)
+                    }
+                }
             }
         }
         finally {
             tmpDir.deleteRecursively()
         }
     }
+
+    fun uploadFileHasNoChanges(resourceId: String, newDatasetHash: String): Boolean
+    {
+        val adr = adrClientBuilder.build()
+        val response = adr.get("resource_show?id=${resourceId}")
+        if (response.statusCode.is2xxSuccessful)
+        {
+            val parser = JSONParser()
```

Comment:
We're using `objectMapper` elsewhere in this controller to read JSON, probably best to be consistent.

Reply:
ObjectMapper() for some reason was not return any data, so after too many hours on it I changed to JsonParser().